packages/consensus/src/types/transaction.cairo

```diff
@@ -123,6 +129,13 @@ pub struct TxOut {
     pub cached: bool,
 }
 
+#[generate_trait]
+pub impl OutPointImpl of OutPointTrait {
+    fn hash(self: @OutPoint) -> felt252 {
```

Comment:
Ok, almost forgot :) This has to take block hash as an extra argument and append to hash state (for hardening against collisions), that's why we introduced it in the block header.